controllers/datadogagent/clusteragent.go

```diff
@@ -661,10 +672,19 @@ func getEnvVarsForClusterAgent(logger logr.Logger, dda *datadoghqv1alpha1.Datado
 		Value: *spec.ClusterAgent.Config.LogLevel,
 	})
 
-	envVars = append(envVars, corev1.EnvVar{
-		Name:      datadoghqv1alpha1.DDAPIKey,
-		ValueFrom: getAPIKeyFromSecret(dda),
-	})
+	// This triggers use of the secret backend.
+	// Otherwise, read from the default or configured secret
+	if secrets.IsEnc(dda.Spec.Credentials.DatadogCredentials.APIKey) {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  datadoghqv1alpha1.DDAPIKey,
+			Value: dda.Spec.Credentials.DatadogCredentials.APIKey,
+		})
```

Comment:
here it means we expose API in an envvar, so anyone that list,get pod can see the api key. 
I would prefer to keep creating a the secret to store the API/APP key. But is is not easy to maintain up-to-date the secret, and redeploy agent when the operator envvar is update (new operator deployment).

maybe we can keep the implementation that you propose and put in our backlog how we can propose something more secure.

Reply:
right, but this would only be stored if the credential set is in the format `ENC[...]` . could that still an issue if a user knows about the SECRET_BACKEND_COMMAND arg?

Reply:
I agree that the secret backend `ENC[...]` is used it is ok to have this value directly as env var.

but, do we need to manage differently the API/APP key and token if it use the secret backend? 
In both case, the secret can contains the the `raw` value or the `ENC["key-secret"]. So for simplicity, I would have keep the value in a secret.

Reply:
i didn't consider that, and it makes sense. thanks, will update to behave this way instead